modules/engine/mrblib/engine.rb

```diff
@@ -42,12 +47,25 @@ def quit
 
       GLFW.default_window_hints
       GLFW.window_hint GLFW::RESIZABLE, GL2::GL_FALSE
-      GLFW.window_hint GLFW::CONTEXT_VERSION_MAJOR, 2
-      GLFW.window_hint GLFW::CONTEXT_VERSION_MINOR, 1
+      GLFW.window_hint GLFW::CONTEXT_VERSION_MAJOR, 3
+      GLFW.window_hint GLFW::CONTEXT_VERSION_MINOR, 3
+      GLFW.window_hint GLFW::OPENGL_FORWARD_COMPAT, GL2::GL_TRUE # for 3.0
+      GLFW.window_hint GLFW::OPENGL_PROFILE, GLFW::OPENGL_CORE_PROFILE # for 3.0 and on
+
+      begin
+        @window = GLFW::Window.new 800, 600, 'Moon Player'
+      rescue GLFWError
+        GLFW.default_window_hints
```

Comment:
You might want to restore this check: https://github.com/polyfox/moon/blob/4e5ad4d68979ca47c51923011dcc82ef3f4d605f/src/window.cxx#L21

The version comparison is not as accurate.